src/actions/genSale/purchaseDomains.ts

```diff
@@ -0,0 +1,81 @@
+import * as ethers from "ethers";
+import { getSaleStatus } from ".";
+import { GenSale } from "../../contracts/types";
+import { errorCheck } from "../../helpers";
+import { Claim, Mintlist, Maybe, GenSaleStatus } from "../../types";
+
+export const purchaseDomains = async (
+    count: ethers.BigNumber,
+    signer: ethers.Signer,
+    contract: GenSale,
+    mintlist: Mintlist
+): Promise<ethers.ContractTransaction> => {
+    const status: GenSaleStatus = await getSaleStatus(contract);
+
+    errorCheck(
```

Comment:
There's a small issue with errorCheck function that's fixed in my branch but not here. errorCheck should throw a `new Error(errorMessage)` instead of just the error message. Otherwise JS will intepret these exceptions as a runtime error.

Reply:
edit, nvm it looks like it's fixed already :)